doc/user_guide/dynamo0p3.rst

```diff
@@ -1388,6 +1410,15 @@ checks (when generating the PSy layer) that any kernels which read
 operator values do not do so beyond the level-1 halo. If any such
 accesses are found then PSyclone aborts.
 
+.. _lfric-array-sizes:
+
+Array sizes
+^^^^^^^^^^^
+
+The size of a :ref:`scalar array <lfric-array>` is described by ``NRANKS*<n>``,
+where *n > 0* is the number of Fortran ranks representing the dimension of the
+array.
+
 .. _lfric-function-space:
```

Comment:
The rules for the arguments to a Kernel subroutine also need to be updated. Presumably, an Array argument will need to be followed by `ndims` and then possibly an integer, rank-1 array of size `ndims` containing the size of each dimension? Or, will `ndims` always be a literal (known) value such that we can each of the dimension sizes as an explicit subroutine argument? (This will allow us to explicitly declare the shape of the `array` argument in the PSy layer which is something we always aim to do.)

Reply:
Yes, we need to pass `ndims` and rank-1 array containing sizes. The proposed design is in the description of #1312, together with examples of code for Kernel, Algorithm and PSy layers (all of them are compilable as far as I remember).